pygpu/collectives.pyx

```diff
@@ -23,12 +23,14 @@ cdef class GpuCommCliqueId:
             res = self.comm_id.internal[:GA_COMM_ID_BYTES]
             return res
 
-        def __set__(self, unsigned char[:] bytearr):
+        def __set__(self, bytearr):
             cdef int length
-            length = bytearr.shape[0]
+            length = len(bytearr)
             if length < GA_COMM_ID_BYTES:
                 raise ValueError, "GpuComm clique id must have length %d bytes" % (GA_COMM_ID_BYTES)
```

Comment:
Check the length after casting to bytes otherwise it can still be too big after.